plugins/carla/cmake/water.cmake

```diff
@@ -0,0 +1,48 @@
+# base config
+set(carla_water_basedir carla/source/modules/water)
+set(carla_water_extra_libs ${carla_pthread_libs})
```

Comment:
Same as above - do not carry obscure variables around. Create INTERFACE libraries that carry usage requirements and link them.